pkg/provider/azure_loadbalancer_accesscontrol.go

```diff
@@ -106,3 +106,53 @@ func (az *Cloud) listSharedIPPortMapping(
 
 	return rv, nil
 }
+
+func (az *Cloud) listAvailableSecurityGroupDestinations(ctx context.Context) ([]netip.Addr, error) {
+	services, err := az.serviceLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("list all services: %w", err)
+	}
+
+	nodes, err := az.nodeLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("list all nodes: %w", err)
+	}
+
+	var rv []netip.Addr
+	for _, svc := range services {
+		// Add additional public IPs
+		{
+			ips, err := loadbalancer.AdditionalPublicIPs(svc)
+			if err == nil {
+				rv = append(rv, ips...)
+			}
+		}
+
+		// Add ingress IPs
+		{
+			for _, ing := range svc.Status.LoadBalancer.Ingress {
+				ip, err := netip.ParseAddr(ing.IP)
+				if err == nil {
+					rv = append(rv, ip)
+				}
+			}
+		}
+	}
+
+	// Add backend node IPs
+	{
+		for _, node := range nodes {
+			for _, addr := range node.Status.Addresses {
+				if addr.Type != v1.NodeInternalIP {
+					continue
+				}
+				ip, err := netip.ParseAddr(addr.Address)
+				if err == nil {
+					rv = append(rv, ip)
+				}
+			}
+		}
+	}
```

Comment:
It'll keep only 
- `additional IPs from annotations`
- `LB service ingress IP`
- `node internal IPs`

in the security rule's destination prefixes. the rest will be removed from managed security rules.

Am I missing any other IPs?